controllers/utils.go

```diff
@@ -159,38 +177,31 @@ func getHMetaAddr(hdb *hapi.HStreamDB) (string, error) {
 		hmetaAddr = hdb.Spec.ExternalHMeta.GetAddr()
 	} else {
 		svc := internal.GetService(hdb, hapi.ComponentTypeHMeta)
-		flags := internal.FlagSet{}
-		if err := flags.Parse(hdb.Spec.HMeta.Container.Args); err != nil {
-			err = fmt.Errorf("parse hmeta args failed. %w", err)
+		port, err := getHMetaContainerPorts(hdb.Spec.HMeta.Container.Args)
+		if err != nil {
 			return "", err
 		}
-		parsedArgs := flags.Flags()
-		port, ok := parseHMetaPort(parsedArgs)
-		if !ok {
-			port = strconv.Itoa(int(hmetaPorts[0].ContainerPort))
-		}
-		hmetaAddr = svc.Name + "." + svc.Namespace + ":" + port
+		hmetaAddr = fmt.Sprintf("%s.%s:%d", svc.Name, svc.Namespace, port[0].ContainerPort)
 	}
 	return hmetaAddr, nil
 }
 
-func getHMetaContainerPorts(container *hapi.Container, parsedArgs map[string]string) (ports []corev1.ContainerPort) {
-	if hmetaPort, ok := parseHMetaPort(parsedArgs); ok {
-		return extendPorts(map[string]string{
-			"port": hmetaPort,
-		}, container.Ports, hmetaPorts)
+func getHMetaContainerPorts(args []string) ([]corev1.ContainerPort, error) {
+	flags := internal.FlagSet{}
+	if err := flags.Parse(args); err != nil {
+		return nil, err
 	}
-	return mergePorts(hmetaPorts, container.Ports)
-}
-
-func parseHMetaPort(parsedArgs map[string]string) (port string, ok bool) {
-	addr := parsedArgs["http-addr"]
-	if addr == "" {
-		return
+	if addr, ok := flags.Flags()["--http-addr"]; ok {
+		if slice := strings.Split(addr, ":"); len(slice) == 2 {
+			return []corev1.ContainerPort{
```

Comment:
Why do we ignore other ports in `container.Ports` when the user defines `--http-addr`?